driver/src/main/com/mongodb/reactivestreams/client/internal/GridFSUploadPublisherImpl.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.reactivestreams.client.internal;
+
+import com.mongodb.reactivestreams.client.Success;
+import com.mongodb.reactivestreams.client.gridfs.GridFSUploadPublisher;
+import org.bson.BsonValue;
+import org.bson.types.ObjectId;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import java.nio.ByteBuffer;
+
+@SuppressWarnings("deprecation")
+public class GridFSUploadPublisherImpl implements GridFSUploadPublisher<Success> {
```

Comment:
In 4.x this class will wrap the async GridFSUploadStream but for now we can share code by wrapping the deprecated local stream.